src/lib/kw_config_loader.sh

```diff
@@ -330,7 +330,14 @@ function load_configuration()
   target_array_local="${target_array}_local"
 
   target_config_file="${target_config}.config"
-  parse_configuration "${KW_ETC_DIR}/${target_config_file}" "$target_array" "$target_array_global"
+
+  # Check if the target configuration file already exists in the expected locations
```

Comment:
I think the name `kw_config_loader` misled you into believing it is a file for dealing with kernel config files, which is false. This file is dedicated to dealing with kw configuration files.

The right place for this check would be in the `src/kernel_config_manager.sh`.